api/tests/test_api.py

```diff
@@ -156,6 +156,19 @@ def test_api_create_datasource(client):
     assert response.status_code == 201
 
 
+def test_api_put_datasource(client):
```

Comment:
It is ok to have this test but I did expect a test that would do a bit more than just return a 204. but honestly not sure how to test since sas token  it si never returned.